server/src/main/java/io/unitycatalog/server/service/iceberg/MetadataService.java

```diff
@@ -18,6 +19,7 @@ public TableMetadata readTableMetadata(String metadataLocation) {
     // TODO: cache fileIO
     FileIO fileIO = fileIOFactory.getFileIO(metadataLocationUri);
 
-    return TableMetadataParser.read(fileIO, metadataLocation);
+    return CompletableFuture.supplyAsync(() -> TableMetadataParser.read(fileIO, metadataLocation))
```

Comment:
qq: since you are blocking for completion anyways, what's the use of using a `CompletableFuture` ? just curious if I'm missing something.

Reply:
Are you trying to throttle the number of threads that are reading the metadata?

Reply:
no, armeria doesn't allow blocking io calls on main thread, this gets around that. In a follow-up PR we can look at cleaning this up a little bit.